### PR TITLE
Protocol list support

### PR DESCRIPTION
**Pull request by acdemyers**

### CODE REVIEW

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 1]
```diff
@@ -660,4 +660,15 @@ ssl.update.certificate.error.useraction=Ensure the location is accessible by the
 ssl.certificate.add.CWPKI0830I=CWPKI0830I: Certificate with the {0} SubjectDN from the {1} environment variable is being added to the {2} keystore.
 ssl.certificate.add.CWPKI0830I.explanation=A certificate was retrieved from an environment variable in the configuration and is being added to the keystore file.
 ssl.certificate.add.CWPKI0830I.useraction=No action is required.
-# -------------------------------------------------------------------------------------------------
\ No newline at end of file
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0831E=CWPKI0831E: The {0} SSL/TLS protocol can not be used.  Extended error is: {1}
```
Comment (helyarp) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Please use "cannot" (1 word) rather than "can not" (2 words) throughout these new messages.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 2]
```diff
@@ -660,4 +660,15 @@ ssl.update.certificate.error.useraction=Ensure the location is accessible by the
 ssl.certificate.add.CWPKI0830I=CWPKI0830I: Certificate with the {0} SubjectDN from the {1} environment variable is being added to the {2} keystore.
 ssl.certificate.add.CWPKI0830I.explanation=A certificate was retrieved from an environment variable in the configuration and is being added to the keystore file.
 ssl.certificate.add.CWPKI0830I.useraction=No action is required.
-# -------------------------------------------------------------------------------------------------
\ No newline at end of file
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0831E=CWPKI0831E: The {0} SSL/TLS protocol can not be used.  Extended error is: {1} 
+ssl.protocol.error.CWPKI0831E.explanation:The protocol provided can not be used.  There is an error when trying to get a SSLContext instance with the protocol value.
```
Comment (helyarp) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Suggested rewrite of the explanation to simplify the wording and increase the quality check scores:

The protocol cannot be used. An error occurs when trying to get a SSLContext instance with the protocol value.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 3]
```diff
@@ -660,4 +660,15 @@ ssl.update.certificate.error.useraction=Ensure the location is accessible by the
...
+ssl.protocol.error.CWPKI0831E=CWPKI0831E: The {0} SSL/TLS protocol can not be used.  Extended error is: {1} 
+ssl.protocol.error.CWPKI0831E.explanation:The protocol provided can not be used.  There is an error when trying to get a SSLContext instance with the protocol value. 
+ssl.protocol.error.CWPKI0831E.useraction:Ensure the configuration uses valid SSL/TLS protocol values for this JVM.
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0832E=CWPKI0832E: The {0} SSL/TLS protocol value provided can not be used when configuring a list of SSL protocols values.
+ssl.protocol.error.CWPKI0832E.explanation:The protocol value is not valid in a protocol list grouping.
+ssl.protocol.error.CWPKI0832E.useraction:Ensure the list of SSL/TLS protocols only includes values appropriate for a protocol list.
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0833E=CWPKI0833E: There is error in the {0} SSL/TLS configuration.  This will prevent the SSL/TLS configuration from being created.
```
Comment (helyarp) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Suggested rewrite of the message to remove the "There is" and "will":

The {0} SSL/TLS configuration has an error that prevents creation of the SSL/TLS configuration.

Please make similar changes to the explanation.

**Review by helyarp — COMMENTED** [review]
Added suggested changes.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 1 above)
Comment (acdemyers) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: @helyarp  The message has be updated please review again.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 2 above)
Comment (acdemyers) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: @helyarp The message has been updated please review again.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 3 above)
Comment (acdemyers) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: @helyarp The message has been updated please review again.

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
@dmuelle I made updates per Pam's suggestions is there anything else needed.  I see you marked id waiting on development.

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 4]
```diff
@@ -660,4 +660,15 @@ ssl.update.certificate.error.useraction=Ensure the location is accessible by the
 ssl.certificate.add.CWPKI0830I=CWPKI0830I: Certificate with the {0} SubjectDN from the {1} environment variable is being added to the {2} keystore.
 ssl.certificate.add.CWPKI0830I.explanation=A certificate was retrieved from an environment variable in the configuration and is being added to the keystore file.
 ssl.certificate.add.CWPKI0830I.useraction=No action is required.
-# -------------------------------------------------------------------------------------------------
\ No newline at end of file
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0831E=CWPKI0831E: The {0} SSL/TLS protocol cannot be used.  Extended error is: {1}
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops:664]: Reviewed Messages:

ssl.protocol.error.CWPKI0831E=CWPKI0831E: The {0} SSL/TLS protocol cannot be used.  Extended error is: {1} 
ssl.protocol.error.CWPKI0831E.explanation: The protocol that is provided cannot be used. An error occurs on trying to get an SSLContext instance with the protocol value.
ssl.protocol.error.CWPKI0831E.useraction: Ensure that the configuration uses valid SSL/TLS protocol values for this JVM.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 5]
```diff
@@ -660,4 +660,15 @@ ssl.update.certificate.error.useraction=Ensure the location is accessible by the
...
 ssl.certificate.add.CWPKI0830I.useraction=No action is required.
-# -------------------------------------------------------------------------------------------------
\ No newline at end of file
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0831E=CWPKI0831E: The {0} SSL/TLS protocol cannot be used.  Extended error is: {1} 
+ssl.protocol.error.CWPKI0831E.explanation:The protocol provided cannot be used.  An error occurs when trying to get a SSLContext instance with the protocol value. 
+ssl.protocol.error.CWPKI0831E.useraction:Ensure the configuration uses valid SSL/TLS protocol values for this JVM.
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0832E=CWPKI0832E: The {0} SSL/TLS protocol value provided cannot be used when configuring a list of SSL protocols values.
+ssl.protocol.error.CWPKI0832E.explanation:The protocol value is not valid in a protocol list grouping.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops:669]: Reviewed messages:

ssl.protocol.error.CWPKI0832E=CWPKI0832E: The {0} SSL/TLS protocol value that is provided cannot be used for configuring a list of SSL protocols values.
ssl.protocol.error.CWPKI0832E.explanation: The protocol value is not valid in a protocol list grouping.
ssl.protocol.error.CWPKI0832E.useraction: Ensure that the list of SSL/TLS protocols includes only values that are appropriate for a protocol list.

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops [hunk 6]
```diff
@@ -660,4 +660,15 @@ ssl.update.certificate.error.useraction=Ensure the location is accessible by the
...
+ssl.protocol.error.CWPKI0832E.useraction:Ensure the list of SSL/TLS protocols only includes values appropriate for a protocol list.
+# -------------------------------------------------------------------------------------------------
+ssl.protocol.error.CWPKI0833E=CWPKI0833E: The {0} SSL/TLS configuration has an error that prevents creation of the SSL/TLS configuration. 
+ssl.protocol.error.CWPKI0833E.explanation:The SSL configuration attributes has and error that will prevent the SSLContext from getting created.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops:673]: Reviewed Messages:
ssl.protocol.error.CWPKI0833E=CWPKI0833E: The {0} SSL/TLS configuration has an error that prevents creation of the SSL/TLS configuration. 
ssl.protocol.error.CWPKI0833E.explanation: The SSL configuration attributes has an error that prevents the SSLContext from getting created. 
ssl.protocol.error.CWPKI0833E.useraction: Ensure that the SSL configuration attributes are correct.  Review the logs for additional errors associated with SSL configuration attributes.

**Comment by acdemyers** [comment]
#build

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 6 above)
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: Reviewed Message: 
CWPKI0833E.explanation: The SSL configuration attributes has an error that prevents SSLContext creation.

**Comment by acdemyers** [comment]
#build

File: dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops (hunk 6 above)
Comment (acdemyers) [reply dev/com.ibm.ws.ssl/resources/com/ibm/ws/ssl/resources/ssl.nlsprops]: The message file has been updated.  Please review the message file.

**Comment by acdemyers** [comment]
#libbybot

**Review by helyarp — COMMENTED** [review]
Changes approved

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#libbybot

**Review by una-tapa — APPROVED** [review]
Looks good to me. Thanks!

**Comment by skasund** [comment]
L2 message review completed

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#libbybot

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#libbybot

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#build

**Comment by acdemyers** [comment]
#libbybot
